src/core/cs3iface.py

```diff
@@ -63,15 +74,8 @@ def stat(endpoint, fileid, userid, versioninv=1):
     '''Stat a file and returns (size, mtime) as well as other extended info using the given userid as access token.
     Note that endpoint here means the storage id. Note that fileid can be either a path (which MUST begin with /),
     or an id (which MUST NOT start with a /). The versioninv flag is natively supported by Reva.'''
-    if endpoint == 'default':
-        raise IOError('A CS3API-compatible storage endpoint must be identified by a storage UUID')
     tstart = time.time()
-    if fileid[0] == '/':
-        # assume this is a filepath
-        ref = cs3spr.Reference(path=fileid)
-    else:
-        # assume we have an opaque fileid
-        ref = cs3spr.Reference(resource_id=cs3spr.ResourceId(storage_id=endpoint, opaque_id=fileid))
+    ref = getcs3reference(endpoint, fileid)
```

Comment:
Further to my other comment, we should rename `fileid` to `fileref` in the pre-existing code to be semantically accurate.